backend/src/disk/fsck.rs

```diff
@@ -0,0 +1,78 @@
+use std::ffi::OsStr;
+use std::path::Path;
+
+use color_eyre::eyre::eyre;
+use tokio::process::Command;
+use tracing::instrument;
+
+use crate::{Error, ResultExt};
+
+#[derive(Debug, Clone, Copy)]
+pub enum RequiresReboot {
+    True,
+    False,
+}
+impl std::ops::BitOrAssign for RequiresReboot {
+    fn bitor_assign(&mut self, rhs: Self) {
+        *self = match (*self, rhs) {
+            (RequiresReboot::False, RequiresReboot::False) => RequiresReboot::False,
+            _ => RequiresReboot::True,
+        }
+    }
+}
+
+#[instrument]
+pub async fn e2fsck(
+    logicalname: impl AsRef<Path> + std::fmt::Debug,
+    repair: bool,
```

Comment:
Side: I question boolean params.
We could pass in an enum that tranforms to the correct e2fsck_cmd
Pass the command
Use Option, and the default is -p and not the repair side